### DISCUSSION TITLE
There is a way to password protect only new PasteBin and not comments to existing PasteBin?

### DISCUSSION THREAD

**Discussion by PeopleInside** [PrivateBin/PrivateBin#1116] [Q&A]
My install as been set to allow only me to create PasteBin with a request of username and password.
Hum i followed some guide to do that but cannot find where the guide is... (if you know can you remind me the link to the guide?) however when someone try to create a PrivateBin a password will be requested by the server. This is ok, its correct.

Suppose I create a PrivateBin note with open discussions... comments require the same password authentication so this is a small issue as I cannot give my credentials to who open the discussion link.

There is a way to not require password only for comments?
If not can be requested as feature request?

**Comment by elrido** [comment]
All API requests go to the same API endpoint, so in your webserver you'd have to add a condition on the contents of the POST/PUT request. The comment's JSON will always contain a "parentid" key, which does not appear in the paste. Nginx, [for example](https://github.com/david-martin/nginx-request-body-example#nginx-example-for-rewriting-based-on-request-body-content), can be made to change routing behaviour based on the presence or absence of strings inside the post body.

PrivateBin itself is fundamentally designed to operate without any concept of users and user permissions (see the [Threat Model](https://github.com/PrivateBin/PrivateBin/wiki/Threat-Model) for an in-depth analysis). Adding mechanisms for authentication is therefore regarded as out of scope.
